plugins/janus_streaming.c

```diff
@@ -2155,6 +2195,115 @@ struct janus_plugin_result *janus_streaming_handle_message(janus_plugin_session
+		json_t *secret = json_object_get(root, "new_secret");
+		json_t *pin = json_object_get(root, "new_pin");
+		json_t *is_private = json_object_get(root, "new_is_private");
+		json_t *permanent = json_object_get(root, "permanent");
+		gboolean save = permanent ? json_is_true(permanent) : FALSE;
+		if(save && config == NULL) {
+			JANUS_LOG(LOG_ERR, "No configuration file, can't edit mountpoint permanently\n");
+			error_code = JANUS_STREAMING_ERROR_UNKNOWN_ERROR;
+			g_snprintf(error_cause, 512, "No configuration file, can't edit mountpoint permanently");
+			goto plugin_response;
+		}
+		guint64 id_value = json_integer_value(id);
+		janus_mutex_lock(&mountpoints_mutex);
+		janus_streaming_mountpoint *mp = g_hash_table_lookup(mountpoints, &id_value);
+		if(mp == NULL) {
+			janus_mutex_unlock(&mountpoints_mutex);
+			JANUS_LOG(LOG_ERR, "No such mountpoint (%"SCNu64")\n", mp->id);
+			error_code = JANUS_STREAMING_ERROR_NO_SUCH_MOUNTPOINT;
+			g_snprintf(error_cause, 512, "No such mountpoint (%"SCNu64")", mp->id);
+			goto plugin_response;
+		}
+		janus_mutex_lock(&mp->mutex);
+		/* A secret may be required for this action */
+		JANUS_CHECK_SECRET(mp->secret, root, "secret", error_code, error_cause,
+			JANUS_STREAMING_ERROR_MISSING_ELEMENT, JANUS_STREAMING_ERROR_INVALID_ELEMENT, JANUS_STREAMING_ERROR_UNAUTHORIZED);
+		if(error_code != 0) {
+			janus_mutex_unlock(&mp->mutex);
+			janus_mutex_unlock(&mountpoints_mutex);
+			goto plugin_response;
+		}
+		/* Edit the mountpoint properties that were provided */
+		if(desc != NULL && strlen(json_string_value(desc)) > 0) {
+			char *old_description = mp->description;
+			char *new_description = g_strdup(json_string_value(desc));
+			mp->description = new_description;
+			g_free(old_description);
+		}
+		if(is_private)
+			mp->is_private = json_is_true(is_private);
+		/* A secret may be required for this action */
+		JANUS_CHECK_SECRET(mp->secret, root, "secret", error_code, error_cause,
+			JANUS_STREAMING_ERROR_MISSING_ELEMENT, JANUS_STREAMING_ERROR_INVALID_ELEMENT, JANUS_STREAMING_ERROR_UNAUTHORIZED);
+		if(error_code != 0) {
+			janus_mutex_unlock(&mountpoints_mutex);
+			goto plugin_response;
+		}
+		if(secret && strlen(json_string_value(secret)) > 0) {
+			char *old_secret = mp->secret;
+			char *new_secret = g_strdup(json_string_value(secret));
+			mp->secret = new_secret;
+			g_free(old_secret);
+		}
+		if(pin && strlen(json_string_value(pin)) > 0) {
+			char *old_pin = mp->pin;
+			char *new_pin = g_strdup(json_string_value(pin));
+			mp->pin = new_pin;
+			g_free(old_pin);
+		}
+		if(save) {
+			/* This change is permanent: save to the configuration file too
+			 * FIXME: We should check if anything fails... */
+			JANUS_LOG(LOG_VERB, "Modifying streaming mountpoint %"SCNu64" permanently in config file\n", mp->id);
+			janus_mutex_lock(&config_mutex);
+			char cat[BUFSIZ];
+			/* The mountpoint ID is the category */
+			g_snprintf(cat, BUFSIZ, "%"SCNu64, mp->id);
```

Comment:
By testing I found this mistake here. The Streaming plugin works differently from AudioBridge/VideoRoom/TextRoom, and the name of the category is _not_ the ID, but its name. Trying to edit an existing mountpoint (actually leaving everything untouched):

    streaming.send({message: {request: "edit", id: 3, secret: "adminpwd", permanent: true}})

I got this new invalid mountpoint instance in my config file:

    [3]
    description = mu-law file source (music)
    secret = adminpwd

while the actual mountpoint instance I wanted to modify (`file-ondemand-sample`, which has ID 3) was left untouched.

Reply:
Ah, I think I see, so we should use `mp->name` in some cases and I guess `mp->id` in others? Sorry, I didn't check configuration saving as it wasn't something I use, I'll look into it, any suggestions further would be helpful.